### PR TITLE
[SMALLFIX][branch-1.6] Improve the use of EmbeddedChannels

### PR DESCRIPTION
**Pull request by apc999**

### CODE REVIEW

**Comment by apc999** [comment]
@calvinjia as discussed, I don't think `Mockito.spy` approach does not apply here, as it requires a return "value" to mock rather than a functor that may generate a return value.

File: core/common/src/test/java/alluxio/EmbeddedChannels.java [hunk 1]
```diff
@@ -80,5 +103,352 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
...
       }
     }
+
+    /**
+     * The channel pipeline that ensures a {@link LastInboundHandler} instance is always in the
+     * end. This is a wrapper of the given pipeline instance with modified {@link #addLast} which
+     * ensures the invariance.
+     */
+    public class EmbeddedChannelPipeline implements ChannelPipeline {
+      private final ChannelPipeline mDefaultChannelPipeline;
```
Comment (gpang) [inline core/common/src/test/java/alluxio/EmbeddedChannels.java]: NIT: can this just be called `mPipeline`?

Comment (apc999) [reply core/common/src/test/java/alluxio/EmbeddedChannels.java]: fixed

**Comment by apc999** [comment]
@gpang comments addressed. PTAL

**Comment by apc999** [comment]
@calvinjia  PTAL

**Pull request merged**